core/dataset_manager.py

```
@@ -82,8 +82,48 @@ def __init__(self, config_manager):
         assert os.path.isdir(raw_filepath), "The dataset filepath provided is not valid."
         self.rfp = raw_filepath
         self.tfp = None
+        self._validate_data()
 
-    def split_and_transform_data(self, transform_function, split=0.8):
+    def _validate_data(self):
+        """
+        Validate all raw data. As of now, checks that:
+            1. Each dataset has a header. Assumes column names are always
+               string. NOTE: if data is also string, then it is impossible
+               to tell whether file has header or not.
+            2. Each dataset is in a valid CSV format. pandas already
+               performs this validation when it reads in CSV files, so just
+               return the error from pandas if reading fails.
+        """
+        format_message = ("The file {file} in folder {folder} was improperly "
+                          "formatted. Please refer to the following error "
+                          "message from pandas for more information: {message}")
+        header_message = ("No header has been provided in file {file} in "
+                          "folder {folder}")
+        for folder in os.listdir(self.rfp):
+            folder_path = os.path.join(os.path.abspath(self.rfp), folder)
+            if not os.path.isdir(folder_path): continue
+            files = os.listdir(folder_path)
+            for file in files:
+                if not file.endswith(".csv"): continue
+                if file[:2] == 'md': continue
+                file_path = os.path.join(folder_path, file)
+                try:
+                    dataset = pd.read_csv(file_path, index_col=False)
+                except Exception as e:
+                    logging.error(str(e))
+                    assert False, format_message.format(
```

Can we raise an exception instead?

---

Sure.